src/qt/pivx/sendconfirmdialog.cpp

```diff
@@ -93,7 +93,10 @@ void TxDetailDialog::setData(WalletModel *model, QModelIndex &index){
             ui->textSend->setText(QString::number(tx->vout.size()) + " recipients");
         }
         ui->textInputs->setText(QString::number(tx->vin.size()));
-        ui->textConfirmations->setText(QString::number(tx->GetDepthInMainChain()));
+        {
+            LOCK(cs_main);
+            ui->textConfirmations->setText(QString::number(tx->GetDepthInMainChain()));
```

Comment:
Good catch with the segfault 👍 .

 We could get it in another way, use the `depth` value that its in the `status` object field in the `TransactionRecord` object.`rec->status.depth`.

We must prevent any LOCK usage anywhere in the UI (and well, in most of the code too if its possible). There should always be another way. Not only for the UI freeze, for the miner too that needs it and its performance will be degraded if it cannot acquire it.